app/src/main/java/org/liberty/android/fantastischmemo/ui/RecentListFragment.java

```diff
@@ -303,24 +325,29 @@ public void onBindViewHolder(final ViewHolder holder, final int position) {
             holder.itemView.setOnClickListener(new View.OnClickListener() {
                 @Override
                 public void onClick(View v) {
-                    Intent myIntent = new Intent();
-                    myIntent.setClass(context, StudyActivity.class);
-                    String dbPath = currentItem.dbPath;
-                    myIntent.putExtra(StudyActivity.EXTRA_DBPATH, dbPath);
-                    recentListUtil.addToRecentList(dbPath);
-                    context.startActivity(myIntent);
+                    if (getSelectedItemCount() == 0) {
+                        Intent myIntent = new Intent();
+                        myIntent.setClass(context, StudyActivity.class);
+                        String dbPath = currentItem.dbPath;
+                        myIntent.putExtra(StudyActivity.EXTRA_DBPATH, dbPath);
+                        recentListUtil.addToRecentList(dbPath);
+                        context.startActivity(myIntent);
+                    } else {
+                        toggleSelection(position);
+                    }
+                    updateActionMode();
                 }
             });
 
             holder.itemView.setOnLongClickListener(new View.OnLongClickListener() {
                 @Override
                 public boolean onLongClick(View v) {
-                    String dbPath = currentItem.dbPath;
-                    DialogFragment df = new OpenActionsFragment();
-                    Bundle b = new Bundle();
-                    b.putString(OpenActionsFragment.EXTRA_DBPATH, dbPath);
-                    df.setArguments(b);
-                    df.show(((FragmentActivity)context).getSupportFragmentManager(), "OpenActions");
+                    AppCompatActivity appCompatActivity = (AppCompatActivity) context;
```

Comment:
I finally got some time to look into this and try to solve this problem, and I found it hard to solve it elegantly. 

I tried to go with route 2, as it introduces no extra dependencies. 

But as my `ActionModeCallback` requires also have access to the `recentListAdapter`, I would have to do it in this way (please correct me if I got it wrong):

1. Instantiate the `ActionModeCallback` in `RecentListAdapter`, and send signal in the `setOnClickListener`
2. Make the `actionModeCallback` a `Parcelable`, and send it with the `Intent`
3. In the `AnyMemo` `Activity`, all the `startSupportActionMode(actionModeCallback)` by using the `ActionModeCall`within the `Intent`.

Or I could do it in this way:

1. Send the signal for ActionMode in `setOnClickListener` of `RecentListAdapter
2. Create an `ActionModeCallBack` in `AnyMemo` activity, `AnyMemo` `Activity` invokes `startSupportActionMode(actionModeCallback)`
3. Send the signal when the selection should be removed or cleared in `ActionModeCallBack`.
4. Add another two two `BroadCastReceiver` in the `RecentListAdapter` to handle the `selectionRemoved()` and `clearSelection()` respectively.

I would slightly prefer the second way. What do you think?